File: NewVersion/JavaScript/dakutenConv.js
function dakuten(arg){
    return [...arg].map((a) => {
        const normalize = a.normalize("NFD")
        return normalize.length > 1 ? `${normalize.substring(0, 1)}゛` : `${normalize}゛`;
    }).join('')
}

console.log(dakuten("だくてんテスト"));

File: NewVersion/Python/dakutenConv.py
import unicodedata

def dakuten(text: str) -> str:
    def conv(a: str) -> str:
        normalize = unicodedata.normalize("NFD", a)
        return f"{normalize[:1]}゛" if len(normalize) > 1 else f"{normalize}゛"
    
    return "".join(map(lambda a: conv(a), list(text)))

if __name__ == '__main__':
    print(dakuten("だくてんテスト"))
